samcli/lib/hook/hook_wrapper.py

```diff
@@ -0,0 +1,176 @@
+"""
+Hooks Wrapper Class
+"""
+import logging
+import importlib
+from pathlib import Path
+from typing import Optional, Dict, cast
+
+
+from .hook_config import HookPackageConfig
+from .exceptions import (
+    InvalidHookPackageConfigException,
+    InvalidHookWrapperException,
+    HookPackageExecuteFunctionalityException,
+)
+
+
+LOG = logging.getLogger(__name__)
+
+
+class IacHookWrapper:
+    """IacHookWrapper
+
+    An IacHookWrapper instance, upon instantiation, looks up the hook package with the specified hook package ID.
+    It provides the "prepare" method, which generates an IaC metadata and output the location of the metadata file.
+
+    Example:
+    ```
+    hook = IacHookWrapper("terraform")
+    metadata_loc = hook.prepare("path/to/iac_project", "path/to/output", True, "path/to/logs")
+    ```
+    """
+
+    _config: Optional[HookPackageConfig]
+
+    _INTERNAL_PACKAGES_ROOT = Path(__file__).parent / ".." / ".." / "hook_packages"
+
+    def __init__(self, hook_package_id: str):
+        """
+        Parameters
+        ----------
+        hook_package_id: str
+            Hook package ID
+        """
+        self._config = None
+        self._load_hook_package(hook_package_id)
+
+    def prepare(
+        self,
+        output_dir_path: str,
+        iac_project_path: Optional[str] = None,
+        debug: bool = False,
+        logs_path: Optional[str] = None,
```

Comment:
no need for this parameter now, as hooks will be called as python libs. Even in the future, we agreed to use the error stream for logs, and output stream for output communication.